File: 01/c++/part2.cpp
#include <stdlib.h>
#include <stdio.h>

#define INPUT "../data/input"

int main() {
    int floor=0;

    // Open file
    FILE* fptr;
    if ((fptr = fopen(INPUT, "r")) == NULL) {
        fprintf(stderr, "Error opening file \"%s\"\n", INPUT);
        exit(EXIT_FAILURE);
    }

    // Read each char in file
    char c;
    int index = 0;
    while ((c = fgetc(fptr)) != EOF) {
        if (c == '(') {
            ++floor;
        } else if (c == ')') {
            --floor;
        }

        ++index;

        if (floor == -1) {
            break;
        }
    }

    fclose(fptr);

    fprintf(stdout, "Got to the basement at character %i\n", index);

    return 0;
}

File: 01/python/part2.py
#!/usr/bin/env python

file = open("../data/input", "r")
symbols = file.read()

floor = 0
symbolIndex = 0

for symbol in symbols:
    symbolIndex += 1

    if (symbol == '('):
        floor += 1
    elif (symbol == ')'):
        floor -= 1

    if (floor == -1):
        break

print("Got to the basement at character", symbolIndex)
